src/AsmResolver.DotNet/TypeDefinition.cs

```diff
@@ -42,6 +42,81 @@ public class TypeDefinition :
         private IList<InterfaceImplementation>? _interfaces;
         private IList<MethodImplementation>? _methodImplementations;
 
+        /// <summary>
+        /// Initialize New Delegate.
+        /// </summary>
+        /// <param name="ns">Delegate Namespace.</param>
+        /// <param name="name">Delegate Name.</param>
+        /// <param name="module">Delegate Module.</param>
+        /// <param name="returnType">Invoke Return Type.</param>
+        /// <param name="parameterTypes">Invoke Parameters Types.</param>
+        /// <param name="attributes">Accessibility Attributes.</param>
+        /// <returns>Fully Implemented Delegate.</returns>
+        public static TypeDefinition CreateDelegate(string ns, string name, ModuleDefinition module, TypeSignature returnType, IEnumerable<TypeSignature> parameterTypes, TypeAttributes attributes = TypeAttributes.NotPublic) =>
```

Comment:
Order of these parameters is a bit inconsistent with the rest of the library.

I think we should stay consistent with the constructors / factory methods of `TypeDefinition`, `TypeReference` and `MethodSignature`. This also allows us to use `params TypeSignature[]` for the parameter types, which allows for nice syntax when using this method.

Possibly something like:
- `CreateDelegate(ModuleDefinition module, string ns, string name, TypeAttributes attributes, TypeSignature returnType, params TypeSignature[] parameterTypes)`
- `CreateDelegate(ModuleDefinition module, string ns, string name, TypeAttributes attributes, MethodSignature invokeSignature)`.